### ISSUE TITLE
Integers are not being transformed post 3.0.0 (Built in ParameterType)

### ISSUE THREAD

**Issue by crcraig92** [cucumber/cucumber-ruby#1279]
## Summary

ParameterTypes don't seem to be functioning at all in Cucumber 3.1.0

The only Transform my framework relied on in Cucumber 2.99 was the first one in the example documentation:
```ruby
Transform /^(-?\d+)$/ do |number|
  number.to_i
end
```
According to the next line in the documentation this is handled by default, which sounds great.  But it's not working for me at all.  I constantly get "undefined method" for various mathematical operators for some integer that is being treated as a string, i.e. "1":String

## Expected Behavior

When I upgrade to cucumber 3.1.0 from 2.99.0 I should remove my transform.rb file and because my only transform is supposedly built in now, my tests should work as they did before.

## Current Behavior

When I run a test that was passing before I get errors like 
```error
undefined method `-' for "1":String (NoMethodError)
```

## Steps I've Tried

I've tried creating dozens of ParameterTypes for the explicit regular expressions used in the step definition and I've tried moving the ParameterTypes into different files.  Originally, my transform was located in /support/transform.rb  I saw some others posting there examples and transform.rb was located in /step_defintions/transform.rb and that also didn't work.  I've tried putting the ParameterType right above the the step definition in question.  Nothing seems to be working for me.

Just for clarity one ParameterType I tried was 

```ruby
ParameterType(
  name: 'number',
  regexp: /(-?\d+)/,
  transformer: ->(number) { number.to_i }
)
```
Where the step definition I'm hoping to catch looks like
```ruby
Then /^I should see (-?\d+) results$/ do | result_count|
  assert_equal(RESULT_PAGE.get_results, result_count)
end
```
I've tried changing transformer into things more like I see in the documentation
```ruby
  transformer: ->(number) { Integer.new(number) }
```

If I change it to result_count.to_i, it fixes the error.  My issue is, there are numerous locations in my framework where this type of step definition was being handled by the transform so I would have a ton of work to just find all the instances of integers being used as strings and fixing it manually.

Is there some solution to replace that transform?  It seems like you are suggesting it should be built in but it's not working in my case.

## Your Environment

* Version used: 3.1.0
* Operating System and version: Windows 10 Build 15063.877
* Link to your project: Can't, sorry
* Ruby: 2.3.3p222

Note: I'm not using rails, that seems to be assumed if you are doing ruby.

**Comment by crcraig92** [comment]
I am trying to reproduce this in a minimal case framework that I can post and so far it seems like with just cucumber 3.1.0 and ruby 2.3.3p222 I can verify that an integer captured in the step is in fact transformed by default.  I will update when I have a significant reproducible case for you too work from, or if I find the issue in my larger framework.

**Comment by crcraig92** [comment]
Further triage with a coworker shed some light on the issue.  

The capture group ([0-9]+), was not caught as an integer and while I'm almost positive I tried that in the ParameterTypes, I must have overlooked that one iteration.  

Adding:

```ruby
ParameterType(
  name: 'number',
  regexp: /[0-9]+/,
  transformer: ->(num) { num.to_i }
)
```
Fixed the issue.

Also, I did not create the framework and I'm not sure why that was used instead of (\d+).
